federation-js/src/composition/compose.ts

```diff
@@ -762,14 +762,22 @@ export function composeServices(services: ServiceDefinition[]): CompositionResul
 
   schema = lexicographicSortSchema(schema);
 
+  // Now that we have a constructed schema, we can filter this map based on types
+  // that made it into the schema successfully. Types that didn't make it into the
+  // schema will have related composition errors.
+  const filteredTypeNameToFieldDirectivesMap = filterMap(
```

Comment:
Instead of filtering here, wouldn't it be easier to make `addFederationMetadataToSchemaNodes` skip types in `typeNameToFieldDirectivesMap` that don't exist in the schema? That avoids the need for `filterMap` and recreating a `Map` even if nothing gets removed. It seems to me the code [here](https://github.com/apollographql/federation/blob/000323590eb63c425d2bcac6acf9248bad3d33b2/federation-js/src/composition/compose.ts#L636) could use a check.